core/field_note.js

```diff
@@ -0,0 +1,852 @@
+   * @type {!Array.<!Array>}
+   * @private
+   */
+  this.mouseEnterWrappers_ = [];
+
+  /**
+   * A wrapper for the mouse down event on the octave down button.
+   * @type {!Array.<!Array>}
+   * @private
+   */
+  this.octaveDownMouseDownWrapper_ = null;
+
+  /**
+   * A wrapper for the mouse down event on the octave up button.
+   * @type {!Array.<!Array>}
+   * @private
+   */
+  this.octaveUpMouseDownWrapper_ = null;
+};
+goog.inherits(Blockly.FieldNote, Blockly.FieldTextInput);
+
+/**
+ * Inset in pixels of content displayed in the field, caused by parent properties.
+ * The inset is actually determined by the CSS property blocklyDropDownDiv- it is
+ * the sum of the padding and border thickness.
+ */
+Blockly.FieldNote.INSET = 5;
+
+/**
+ * Height of the top area of the field, in px.
+ * @type {number}
+ * @const
+ */
+Blockly.FieldNote.TOP_MENU_HEIGHT = 32 - Blockly.FieldNote.INSET;
+
+/**
+ * Padding on the top and sides of the field, in px.
+ * @type {number}
+ * @const
+ */
+Blockly.FieldNote.EDGE_PADDING = 1;
+
+/**
+ * Height of the drop shadow on the piano, in px.
+ * @type {number}
+ * @const
+ */
+Blockly.FieldNote.SHADOW_HEIGHT = 4;
+
+/**
+ * A translucent gray color, for the shadow on the piano.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.SHADOW_COLOR = 'rgba(.2, .2, .2, .2)';
+
+/**
+ * A color for the white piano keys.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.WHITE_KEY_COLOR = '#FFFFFF';
+
+/**
+ * A color for the black piano keys.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.BLACK_KEY_COLOR = '#323133';
+
+/**
+ * A color for stroke around black piano keys.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.BLACK_KEY_STROKE = '#555555';
+
+/**
+ * A color for the selected state of a piano key.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.KEY_SELECTED_COLOR = '#b0d6ff';
+
+/**
+ * The number of white keys in one octave on the piano.
+ * @type {number}
+ * @const
+ */
+Blockly.FieldNote.NUM_WHITE_KEYS = 8;
+
+/**
+ * Height of a white piano key, in px.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.WHITE_KEY_HEIGHT = 72;
+
+/**
+ * Width of a white piano key, in px.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.WHITE_KEY_WIDTH = 40;
+
+/**
+ * Height of a black piano key, in px.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.BLACK_KEY_HEIGHT = 40;
+
+/**
+ * Width of a black piano key, in px.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.BLACK_KEY_WIDTH = 32;
+
+/**
+ * Radius of the curved bottom corner of a piano key, in px.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.KEY_RADIUS = 6;
+
+/**
+ * Bottom padding for the labels on C keys.
+ * @type {string}
+ * @const
+ */
+Blockly.FieldNote.KEY_LABEL_PADDING = 8;
+
+/**
+ * An array of objects with data describing the keys on the piano.
+ * @type {Array.<{name: String, pitch: Number, isBlack: boolean}>}
+ * @const
+ */
+Blockly.FieldNote.KEY_INFO = [
+  {name: 'C', pitch: 0},
```

Comment:
We should translate them.

Reply:
@apple502j do musical notes have translations in other languages? At least in Chinese, a C is a C.

Reply:
Thanks for your thoughts @apple502j and @Kenny2github - we quickly discussed it with the team here and decided to keep the note names untranslated.